presto-native-execution/presto_cpp/main/operators/BroadcastWrite.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/core/PlanNode.h"
+#include "velox/exec/Operator.h"
+
+namespace facebook::presto::operators {
+
+class BroadcastWriteNode : public velox::core::PlanNode {
+ public:
+  BroadcastWriteNode(
+      const velox::core::PlanNodeId& id,
+      const std::string& basePath,
+      velox::core::PlanNodePtr source)
+      : velox::core::PlanNode(id),
+        basePath_{basePath},
+        sources_{std::move(source)},
+        outputType_{velox::ROW({velox::VARCHAR()})} {}
+
+  folly::dynamic serialize() const override;
+
+  static velox::core::PlanNodePtr create(
+      const folly::dynamic& obj,
+      void* context);
+
+  const velox::RowTypePtr& outputType() const override {
+    return outputType_;
+  }
+
+  const std::vector<velox::core::PlanNodePtr>& sources() const override {
+    return sources_;
+  }
+
+  const std::string& broadcastWriteBasePath() const {
```

Comment:
This name can be shorted to basePath since "broadcastWrite" is already included in the class name.